src/main/groovy/com/ullink/gradle/nunit/NUnit.groovy

```diff
@@ -64,13 +71,86 @@ class NUnit extends ConventionTask {
         project.file(getReportFolder())
     }
 
+    @OutputFile
     File getTestReportPath() {
-        new File(getReportFolderImpl(), reportFileName)
+        // for the non-default nunit tasks, ensure we write the report in a separate file
+        def reportFileNamePrefix = name == 'nunit' ? '' : name
+        new File(getReportFolderImpl(), reportFileNamePrefix + reportFileName)
     }
 
     @TaskAction
     def build() {
-        def cmdLine = [nunitExec.absolutePath, *buildCommandArgs()]
+        decideExecutionPath(this.&singleRunExecute, this.&multipleRunsExecute)
+    }
+
+    def decideExecutionPath(Closure singleRunAction, Closure multipleRunsAction) {
+        if (!test && run) {
```

Comment:
if (!test && run) {
    test = run
}
What does this mean? It's some logic in case nunit2 or nunit3 is used by the plugin?